checks/frozen_deps.go

```diff
@@ -104,33 +152,36 @@ func validateShellScriptIsFreeOfInsecureDownloads(pathfn string, content []byte,
 	return validateShellFile(pathfn, content, dl)
 }
 
-func isDockerfileFreeOfInsecureDownloads(c *checker.CheckRequest) checker.CheckResult {
+func isDockerfileFreeOfInsecureDownloads(c *checker.CheckRequest) (int, error) {
 	r, err := CheckFilesContent2("*Dockerfile*", false, c, validateDockerfileIsFreeOfInsecureDownloads)
-	return createResultForIsDockerfileFreeOfInsecureDownloads(r, err)
+	return createReturnForIsDockerfileFreeOfInsecureDownloads(r, c.Dlogger, err)
 }
 
 // Create the result.
-func createResultForIsDockerfileFreeOfInsecureDownloads(r bool, err error) checker.CheckResult {
+func createReturnForIsDockerfileFreeOfInsecureDownloads(r bool, dl checker.DetailLogger, err error) (int, error) {
```

Comment:
A lot of these createReturn* functions seem exactly the same, apart from the log message. 

It would reduce a lot of duplicate to have a single one which takes a log message as an argument instead.

Reply:
good point. Done.